fastn-core/src/doc.rs

```diff
@@ -25,7 +25,17 @@ fn cached_parse(
     fastn_core::utils::cache_it(id, C { doc, hash }).map(|v| v.doc)
 }
 
-#[tracing::instrument(skip_all)]
+pub fn package_dependent_builtins(
+    pkg: &fastn_core::Package,
+    req_path: &str,
+) -> Vec<(String, fastn_resolved::Definition)> {
```

Comment:
Instead of returning a Vec you can return a fixed size array (and change function definition when the array length changes, eg when you add more items, to try to avoid one malloc, as arrays would be created on the stack.

Reply:
done in https://github.com/fastn-stack/fastn/pull/2057/commits/897fc62b9098dcf1ef6cca4629f8fd5679501c7b